pkg/controller/machinedeployment/controller.go

```diff
@@ -377,6 +379,8 @@ func (r *ReconcileMachineDeployment) MachineSetToDeployments(o handler.MapObject
 		return nil
 	}
 
+	// Check if the controller reference is already set and
+	// return an empty result when one is found.
```

Comment:
Is this actually the behavior that we want? won't this shortcut if any of the machineSets have a controllerRef instead of just excluding the ones that do?

Reply:
Doesn't the other watcher should take care of the non-orphaned ones?

Reply:
It does, but if we return early here because we find a single MachineSet that already has a controller reference, then don't we potentially skip reconciling/adopting other MachineSets that do not have a controller reference?

Reply:
nm, I multitasking had me misreading the loops.

Reply:
Seems the function only handles one MachineSet at a time, unless I'm missing something